### ISSUE TITLE
Chromeperf dashboard UI: Y axis label covers axis tick values

### ISSUE THREAD

**Issue by qyearsley** [catapult-project/catapult#1344]
**Issue by [primi...@chromium.org](https://code.google.com/u/109349271600817121881/)**
_2015-08-17T17:20:42.000Z_
_Originally opened as http://crbug.com/521664_

---

Small UI bug on chromeperf dashboards
See https://chromeperf.appspot.com/report?sid=8f760abd50fdfad2206ba581542ca91d724139e062facf9e081388cbafac7c5a

The text &quot;bytes (lower is better)&quot; partially covers with the actual numbers of the Y axis, making them hard to read.

Can we haz more left margin (or divide more the numbers)
Thanks

**Comment by sophiasorokina** [comment]
Hello, @qyearsley. I am new to contributing and interested in taking some issue. Is this issue relevant? Or maybe I could try another one? It's my first time so I'm kind of afraid to bother you..

**Comment by qyearsley** [comment]
Hi @sophiasorokina; yep, this issue is still relevant, since there are still graphs where the y-values are greater than 1000000.

In the original report, @primiano suggested potentially adding more left margin, but this has the disadvantage of wasting screen space for most of the graphs, where the y-values are less than 1000.

There's another potential solution that would solve this and avoiding space: Reformat the numbers so that they take up less space using abbreviations like K, M, G, etc. The format of the numbers is determined by [formatYAxis in dashboard/elements/chart-container.html](https://github.com/catapult-project/catapult/blob/master/dashboard/dashboard/elements/chart-container.html#L1480).

So, for example, that function could be changed to return strings that are at most 6 characters long, e.g.:
```
    1 -> 1
    123.4567 -> 123.45
    123456 -> 123456
    1234567 -> 1.234M
    12345678 -> 12.34M
    123456789 -> 123.4M
    1234567890 -> 1.234G
```

What do you think?

Don't hesitate to ask any questions :-) When you're ready to upload a patch for review, see [CONTRIBUTING.md](https://github.com/catapult-project/catapult/blob/master/CONTRIBUTING.md).

**Comment by sophiasorokina** [comment]
Thank you, @qyearsley! I think I can manage this. I'll be done today I hope)

**Comment by dave-2** [comment]
Hi, @sophiasorokina, are you still interested in taking on this task?

@qyearsley @eakuefner  It's insufficient to add a suffix to the number. For example, if the units are kilobytes, rephrasing "1234567" as "1.23M" is misleading! It should be "1.23 GB". The upcoming Values system has some unit-aware formatting ability that can handle these cases. (More complex scenario: rephrasing "86417483 ms" as "1.00 days")

I would say, yes, the additional margin is somewhat wasteful of space, but we're generally not so limited by space in this context, so it's a good quick fix for now. (This would be a form of premature optimization, I think.)

**Comment by qyearsley** [comment]
Makes sense to me; I can see how 1.23M kb would be really confusing :-) Another alternative would be scientific notation, e.g.:

    1 -> 1
    123.4567 -> 123.45
    123456 -> 123456
    1234567 -> 1.234e6
    12345678 -> 1.234e7
    123456789 -> 1.234e8
    1234567890 -> 1.234e9

**Comment by primiano** [comment]
1.234e8 bytes would make it even harder to reason about?
I am probably missing some context here. We push values to the dashboard saying that they are "bytes". Why cannot we just do the right thing and know how to scale up bytes on the dashboard (and either do nothing if the unit is already scaled, or do something similar if the unit is KB)?
I assume we know if something is pushed as "bytes" vs "KB", right?

**Comment by anniesullie** [comment]
@primiano I think this is just one of those "not ready yet" things. On the trace-viewer side, @petrcermak has been doing awesome work on [units](https://github.com/catapult-project/catapult/blob/master/tracing/tracing/value/unit.html) in the value system. We are working on refactoring telemetry and the perf dashboard so they all use the common values system shared code and these units "just work" everywhere. At that point we would know how to scale up bytes on the dashboard (and in telemetry results.html, and bisect output, etc) and just do the right thing.

**Comment by benshayden** [comment]
Aside from the Units formatting, this appears to have been fixed in #3519
